src/apps/chifra/internal/receipts/output.go

```diff
@@ -45,12 +65,77 @@ func (opts *ReceiptsOptions) ReceiptsInternal() (err error, handled bool) {
 	if opts.Globals.ApiMode {
 		return nil, false
 	}
+	if opts.Articulate {
+		err = opts.Globals.PassItOn("getReceipts", opts.Globals.Chain, opts.toCmdLine(), opts.getEnvStr())
+		return err, true
+	}
 
-	handled = true
-	err = opts.Globals.PassItOn("getReceipts", opts.Globals.Chain, opts.toCmdLine(), opts.getEnvStr())
-	// EXISTING_CODE
+	notFound := make([]error, 0)
+	defer func() {
+		for _, err := range notFound {
+			fmt.Fprintln(os.Stderr, err)
```

Comment:
This should be reported through standard err (using `logger.Error`) so it doesn't intermix with the data produced. 

The question to ask is this: if a command-line user pipes the output into a file, should the error message be in the file or should it be reported to the end user?

Reply:
Good point about using the package we already have 👍 , but `logger` also uses `os.Stderr`: https://github.com/TrueBlocks/trueblocks-core/blob/master/src/apps/chifra/pkg/logger/logger.go#L72
I think that our whole Go codebase expects an error to be returned and then it all works nice in both environments. Here we have a new situation, where there can be multiple errors. A quick and dirty workaround would be to just return the first error here. A better would be to try to change the console/server code to a generic version that expects `error` OR `[]error`

Reply:
In the C++ code, the errors are reported to the server's console and in an object called `errors` (at the same level in the response as `data`)--so both stderr and in the response (i.e. stdout).

In command-line code, the error is reported only to stderr.

I'm going to resolve this conversation and write an issue.